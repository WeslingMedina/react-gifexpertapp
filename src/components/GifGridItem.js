import react from 'react';
import {PropTypes} from 'prop-types';

const GifGridItem = ({title,url}) => {
    // console.log(id,title,url);
    return (
        //como estamos trabajando en .js no podemos usar la palabra reservada class que es de html, por lo tanto, se decidi√≥ usar className en vez de class
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem;