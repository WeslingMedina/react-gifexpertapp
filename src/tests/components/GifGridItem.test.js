import GifGridItem from "../../components/GifGridItem";
import {shallow} from "enzyme";
describe("testing gifGridItem", () => {
    const data = {
        title : "Un titulo", 
        url : "https://localhost/logo.png"
    }
    const wrapper = shallow(<GifGridItem {...data}/>);

    test("should show the component in a right way",() => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un pÃ¡rrafo con el title del objeto data',() => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(data.title);
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.html());//si colocamos el .html() podemos ver el objeto completo
        // console.log(img.props());//se puede ver las propiedades de la imagen, el src y alt en un objeto
        // console.log(img.prop('src'));//con esto sacamos una propiedad en especifico tambien se puede hacer con el .props().src

        expect(img.prop('src')).toBe(data.url);
        expect(img.prop('alt')).toBe(data.title);
    });

    //evaluando las clases del div principal del hook GifGridItem
    test("debe de tener la clase animate__fadeIn", () => {
        const div = wrapper.find('div');
        expect(div.props().className.includes("animate__fadeIn")).toBe(true);
    });

});