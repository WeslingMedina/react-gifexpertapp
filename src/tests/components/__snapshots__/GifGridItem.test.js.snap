// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing gifGridItem should show the component in a right way 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/logo.png"
  />
  <p>
    Un titulo
  </p>
</div>
`;
